modules/ckanext-ytp_main/ckanext/ytp/commands.py

```diff
@@ -482,52 +482,76 @@ def add_to_groups(ctx, config, dryrun):
-def send_harvester_status_emails(ctx, config, dryrun):
+def send_harvester_status_emails(ctx, config, dryrun, force, all_harvesters):
     load_config(config or ctx.obj['config'])
-    status = get_action('harvester_status')({}, {})
 
-    errored_runs = any(item.get('status') != 'ok' for item in status.values())
-    last_runs = (item.get('last_run') for item in status.values())
-    old_runs = any(_elapsed_since(last_run).days > 2 for last_run in last_runs if last_run is not None)
+    email_notification_recipients = t.aslist(t.config.get('ckanext.ytp.harvester_status_recipients', ''))
+
+    if not email_notification_recipients and not dryrun:
+        print 'No recipients configured'
+        return
+
+    status_opts = {} if not all_harvesters else {'include_manual': True, 'include_never_run': True}
+    status = get_action('harvester_status')({}, status_opts)
 
-    if not (errored_runs or old_runs):
+    errored_runs = any(item.get('errors') != 0 for item in status.values())
+    running = (item.get('started') for item in status.values() if item.get('status') == 'running')
+    stuck_runs = any(_elapsed_since(started).days > 1 for started in running)
+
+    if not (errored_runs or stuck_runs) and not force:
         print 'Nothing to report'
         return
 
-    email_notification_recipients = t.aslist(t.config.get('ckanext.ytp.harvester_status_recipients', ''))
+    if len(status) == 0:
+        print 'No harvesters matching criteria found'
+        return
 
-    if email_notification_recipients:
-        site_title = t.config.get('ckan.site_title', '')
-        today = datetime.now().date().isoformat()
-        status_template = '%%(title)-%ds | %%(last_run)s: %%(status)s' % max(len(title) for title in status)
-
-        msg = '%(site_title)s - Harvester summary %(today)s\n\n%(status)s' % {
-                'site_title': site_title,
-                'today': today,
-                'status': '\n'.join(status_template % {
-                    'title': title,
-                    'last_run': _pretty_time(values.get('last_run')),
-                    'status': values.get('status')
-                    } for title, values in status.items())
+    site_title = t.config.get('ckan.site_title', '')
+    today = datetime.now().date().isoformat()
+
+    status_templates = {
+            'running': '%%(title)-%ds | Running since %%(time)s with %%(errors)d errors',
+            'finished': '%%(title)-%ds | Finished %%(time)s with %%(errors)d errors',
+            'pending': '%%(title)-%ds | Pending since %%(time)s'}
+    unknown_status_template = '%%(title)-%ds | Unknown status: %%(status)s'
+    max_title_length = max(len(title) for title in status)
+
+    def status_string(title, values):
+        template = status_templates.get(values.get('status'), unknown_status_template)
+        status = values.get('status')
+        time_field = 'finished' if status == 'finished' else 'started'
+        return template % max_title_length % {
```

Comment:
What does the max_title_length here do?

Reply:
It adjusts the title column width to be the same for all status rows so they're aligned.

Reply:
Ok nice didn't know that syntax :)